### PR TITLE
Scl binding

### PR DESCRIPTION
**Pull request by juchiast**
Writing scl-serde turned out to be very time consuming. I do something else instead.

Important functions are `to_rlua_value`, `to_scl_value` and `scl_decode`.

### CODE REVIEW

File: src/bindings/scl.rs [hunk 1]
```diff
@@ -80,7 +79,19 @@ fn to_scl_value(lua: &rlua::Lua, val: rlua::Value) -> scl::Value {
 }
 
 fn escape(s: String) -> String {
-    "\"".to_owned() + &s.replace('"', r##"\""##) + "\""
+    if s.contains(r##"""""##) {
+        // escape \n and "
+        // return "\"".to_owned() + &s.replace('"', r##"\""##).replace('\n', "\\n") + "\"";
+        //
+        // this doesn't work, see:
+        // https://github.com/Keats/scl/issues/10
```
Comment (juchiast) [inline src/bindings/scl.rs]: See this issue:          // https://github.com/Keats/scl/issues/10

File: src/bindings/scl.rs [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+        a=1.2,
+        b={c=1},
+        d={day=1,month=2,year=2018},
+        e={2, 3, 4},
+        }
+        local s = scl.from_table(x)
+        print(s)
+        local y = scl.to_table(s)
+        -- don't mess with floating point number
+        -- assert(y.a == 1.2)
```
Comment (Arnaz87) [inline src/bindings/scl.rs]: What's up with this? Why doesn't it work?

Comment (juchiast) [reply src/bindings/scl.rs]: Nothing wrong with it, I just don't want to compare floating point numbers.

Comment (Arnaz87) [reply src/bindings/scl.rs]: why not use 1.5? that's perfectly representable

Comment (juchiast) [reply src/bindings/scl.rs]: I think you can push to my branch, so change what you want about it.

**Pull request merged**